lib/cadet_web/admin_controllers/admin_assessments_controller.ex

```diff
@@ -98,6 +100,15 @@ defmodule CadetWeb.AdminAssessmentsController do
         Map.put(updated_assessment, :max_team_size, max_team_size)
       end
 
+    updated_assessment =
+      if is_nil(has_token_counter) do
+        updated_assessment
+      else
+        updated_assessment
+        |> Map.put(:has_token_counter, has_token_counter)
+        |> Map.put(:has_voting_features, has_voting_features)
+      end
+
```

Comment:
I feel like this should be two separate "if" checks, one for each param&hellip;. Thoughts?

Reply:
Yea, I think that's better. I have changed it to be separated now. Let me know if there's any other issues!